### PR TITLE
Issue 287 - Added spinner call function for backend calls

### PR DESCRIPTION
**Pull request by ahmed-shehata**
Issue #287 

 - Added spinner call function for backend calls
- Added tests for deleting preselected address

### CODE REVIEW

File: AtlasUI/AtlasUI/CheckoutSummaryViewController.swift [hunk 1]
```diff
@@ -54,13 +54,18 @@ class CheckoutSummaryViewController: UIViewController, CheckoutProviderType {
...
+    private func showLoader() {
         self.loaderView.show()
     }
 
-    internal func hideLoader() {
+    private func hideLoader() {
         self.loaderView.hide()
     }
+
+    internal func makeNetworkRequestWithSpinner(requestToMake: (() -> Void) -> Void) {
```
Comment (bonkey) [inline AtlasUI/AtlasUI/CheckoutSummaryViewController.swift]: Naming is too misleading. `() -> Void` doesn't mean it's a network request at all.

How about naming from #138 or anything better but similar in meaning.

Comment (ahmed-shehata) [inline AtlasUI/AtlasUI/CheckoutSummaryViewController.swift]: typealias?

Comment (bonkey) [inline AtlasUI/AtlasUI/CheckoutSummaryViewController.swift]: I mean "makeNetworkRequestWithSpinner". I does **not** make network requests. And actually without spinner, too, but _loader_.

File: AtlasUI/AtlasUI/CheckoutSummaryViewController.swift [hunk 2]
```diff
@@ -62,7 +62,7 @@ class CheckoutSummaryViewController: UIViewController, CheckoutProviderType {
         self.loaderView.hide()
     }
 
-    internal func makeNetworkRequestWithSpinner(requestToMake: (() -> Void) -> Void) {
+    internal func displayLoaderWithRequest(requestToMake: (() -> Void) -> Void) {
```
Comment (bonkey) [inline AtlasUI/AtlasUI/CheckoutSummaryViewController.swift]: I repeat, will the true name here please stand up?

7jemxmtg66hcy

Comment (ahmed-shehata) [inline AtlasUI/AtlasUI/CheckoutSummaryViewController.swift]: done

**Comment by bonkey** [comment]
👍

**Comment by ahmed-shehata** [comment]
👍

**Pull request merged**